C/services/notification/notification_manager.cpp

```diff
@@ -780,6 +780,13 @@ bool NotificationInstance::handleState(bool evalRet)
 	// Calculate time diff
 	timersub(&now_tv, &m_lastSentTv, &diffTimeTv);
 
+	// round up the seconds If microseconds are greater than or equal to 500000
+	if (diffTimeTv.tv_usec >= 500000)
```

Comment:
It would be better if define it as const

Reply:
I do not think, i fully understand the rationality of the logic here. What is the need to look into lower order precision (micros second) ? to override, high order precision ? Does our need to have time precision of 500 milli second ..? 

I think it is better if we stick with the time precisions. 

I see we are using gettimeofday() here are some resons we should not use it https://blog.habets.se/2010/09/gettimeofday-should-never-be-used-to-measure-time.html

Here the example code with custom precision.
```cpp
    auto start = std::chrono::steady_clock::now();

    // Perform some operations
    for (int i = 0; i < 1000000; ++i);

    auto end = std::chrono::steady_clock::now();
    auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - start);
```